### PR TITLE
fix: somalier container

### PR DESCRIPTION
**Pull request by mathiasbio**
## Description

During earlier development branches merged into develop: https://github.com/Clinical-Genomics/BALSAMIC/pull/1524 it was seen that Somalier container build failed with this error:

```
#11 [ 8/10] RUN apk add  --no-cache openblas &&     cd /somalier &&     nim c -d:danger -d:nsb_static -d:release -d:openmp -d:blas=openblas -d:lapack=openblas -o:/usr/bin/somalier src/somalier &&     cp scripts/ancestry-labels-1kg.tsv / &&     rm -rf /somalier && somalier --help
#11 0.082 fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/main/x86_64/APKINDEX.tar.gz
#11 0.265 fetch http://dl-cdn.alpinelinux.org/alpine/v3.11/community/x86_64/APKINDEX.tar.gz
#11 0.486 (1/1) Installing openblas (0.3.7-r3)
#11 0.941 OK: 677 MiB in 96 packages
#11 1.038 Hint: used config file '/nim-1.6.20/config/nim.cfg' [Conf]
#11 1.038 Hint: used config file '/nim-1.6.20/config/config.nims' [Conf]
#11 1.038 Hint: used config file '/somalier/nim.cfg' [Conf]
#11 1.040 ......................................................................................................................................
#11 1.806 /somalier/src/somalierpkg/estimate_contamination.nim(114, 7) Hint: 'sites_used' is declared but not used [XDeclaredButNotUsed]
#11 1.806 /somalier/src/somalierpkg/estimate_contamination.nim(112, 6) Hint: 'estimate_contamination' is declared but not used [XDeclaredButNotUsed]
#11 1.806 /somalier/src/somalierpkg/estimate_contamination.nim(34, 7) Hint: 'error_rate' is declared but not used [XDeclaredButNotUsed]
#11 1.806 /somalier/src/somalierpkg/relate.nim(35, 33) Warning: type pragmas follow the type name; this form of writing pragmas is deprecated [Deprecated]
#11 1.886 /somalier/src/somalierpkg/relate.nim(702, 9) Hint: 's' is declared but not used [XDeclaredButNotUsed]
#11 1.975 /somalier/src/somalierpkg/relate.nim(19, 8) Warning: imported and not used: 'estimate_contamination' [UnusedImport]
#11 1.975 ......
#11 2.020 /somalier/src/somalierpkg/findsites.nim(57, 9) Hint: 'x' is declared but not used [XDeclaredButNotUsed]
#11 2.050 /somalier/src/somalierpkg/findsites.nim(173, 13) Hint: 'x' is declared but not used [XDeclaredButNotUsed]
#11 2.073 /somalier/src/somalierpkg/findsites.nim(273, 9) Hint: 'x' is declared but not used [XDeclaredButNotUsed]
#11 2.079 .......
#11 2.123 /root/.nimble/pkgs/nimblas-0.3.1/nimblas/private/common.nim(41, 12) Warning:     The flag -d:openblas is deprecated since NimBLAS 0.2
#11 2.123 Use -d:blas=LIBNAME instead
#11 2.123    [User]
#11 2.124 /root/.nimble/pkgs/nimblas-0.3.1/nimblas/private/common.nim(52, 7) Hint: Using BLAS library matching pattern: libopenblas.so(|.3|.2|.1|.0) [User]
#11 2.124 ...................
#11 2.180 /root/.nimble/pkgs/nimcuda-0.2.2/nimcuda/cuda12_5/libpaths.nim(146, 10) Error: Could not find the cuda source headers! Please specify the location of the cuda includes directory by passing `-d:CudaIncludes="YOUR_PATH"` to the nim compiler.
#11 ERROR: process "/bin/sh -c apk add  --no-cache openblas &&     cd /somalier &&     nim c -d:danger -d:nsb_static -d:release -d:openmp -d:blas=openblas -d:lapack=openblas -o:/usr/bin/somalier src/somalier &&     cp scripts/ancestry-labels-1kg.tsv / &&     rm -rf /somalier && somalier --help" did not complete successfully: exit code: 1
------
 > [ 8/10] RUN apk add  --no-cache openblas &&     cd /somalier &&     nim c -d:danger -d:nsb_static -d:release -d:openmp -d:blas=openblas -d:lapack=openblas -o:/usr/bin/somalier src/somalier &&     cp scripts/ancestry-labels-1kg.tsv / &&     rm -rf /somalier && somalier --help:
1.975 /somalier/src/somalierpkg/relate.nim(19, 8) Warning: imported and not used: 'estimate_contamination' [UnusedImport]
1.975 ......
2.020 /somalier/src/somalierpkg/findsites.nim(57, 9) Hint: 'x' is declared but not used [XDeclaredButNotUsed]
...
2.123 /root/.nimble/pkgs/nimblas-0.3.1/nimblas/private/common.nim(41, 12) Warning:     The flag -d:openblas is deprecated since NimBLAS 0.2
2.123 Use -d:blas=LIBNAME instead
2.123    [User]
2.124 /root/.nimble/pkgs/nimblas-0.3.1/nimblas/private/common.nim(52, 7) Hint: Using BLAS library matching pattern: libopenblas.so(|.3|.2|.1|.0) [User]
2.124 ...................
2.180 /root/.nimble/pkgs/nimcuda-0.2.2/nimcuda/cuda12_5/libpaths.nim(146, 10) Error: Could not find the cuda source headers! Please specify the location of the cuda includes directory by passing `-d:CudaIncludes="YOUR_PATH"` to the nim compiler.
------
BALSAMIC/containers/somalier/Dockerfile:62
--------------------
  61 |     
  62 | >>> RUN apk add  --no-cache openblas && \
  63 | >>>     cd /somalier && \
  64 | >>>     nim c -d:danger -d:nsb_static -d:release -d:openmp -d:blas=openblas -d:lapack=openblas -o:/usr/bin/somalier src/somalier && \
  65 | >>>     cp scripts/ancestry-labels-1kg.tsv / && \
  66 | >>>     rm -rf /somalier && somalier --help
  67 |     
--------------------
ERROR: failed to solve: process "/bin/sh -c apk add  --no-cache openblas &&     cd /somalier &&     nim c -d:danger -d:nsb_static -d:release -d:openmp -d:blas=openblas -d:lapack=openblas -o:/usr/bin/somalier src/somalier &&     cp scripts/ancestry-labels-1kg.tsv / &&     rm -rf /somalier && somalier --help" did not complete successfully: exit code: 1
Error: buildx failed with: ERROR: failed to solve: process "/bin/sh -c apk add  --no-cache openblas &&     cd /somalier &&     nim c -d:danger -d:nsb_static -d:release -d:openmp -d:blas=openblas -d:lapack=openblas -o:/usr/bin/somalier src/somalier &&     cp scripts/ancestry-labels-1kg.tsv / &&     rm -rf /somalier && somalier --help" did not complete successfully: exit code: 1

```

#### Changed

- pinned arraymancer version in somalier docker build recipe

## Documentation

- [x] N/A
- [ ] Updated Balsamic documentation to reflect the changes as needed for this PR.
    - [Document Name]

### Feature Tests

- [x] After pulling down the fixed somalier container from dockerhub it is working. Pulled down container: 

image

And ran somalier successfully:

image

### Pipeline Integrity Tests

- **Report deliver (generation of the `.hk` file)**
    - [x] N/A
    - [ ] Verified
- **TGA T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **TGA T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **UMI T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/O Workflow**
    - [x] N/A
    - [ ] Verified
- **WGS T/N Workflow**
    - [x] N/A
    - [ ] Verified
- **QC Workflow**
    - [x] N/A
    - [ ] Verified
- **PON Workflow**
    - [x] N/A
    - [ ] Verified

### Documentation

- **Atlas documentation**
    - [x] N/A
    - [ ] Updated: [Link]
- **Web portal for Clinical Genomics**
    - [x] N/A
    - [ ] Updated: [Link]

### Panel of Normal specific criteria

- [ ] The PR includes the addition of a new Panel of Normals
- [ ] The samples have been verified to adhere to the sample selection criteria on [Atlas PoN creation instructions for Balsamic](https://atlas.scilifelab.se/infrastructure/BALSAMIC/balsamic_pon/#procedure)

### User Changes

- [x] N/A
- [ ] This PR affects the output files or results.
    - [ ] User feedback is considered unnecessary because [Justification].
    - [ ] Affected users have been included in the development process and given a chance to provide feedback.

### Infrastructure Changes

- **Stored files in Housekeeper**
    - [x] N/A
    - [ ] Updated: [Link]
- **CG (CLI and delivered/uploaded files)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Servers (configuration files on Hasta)**
    - [x] N/A
    - [ ] Updated: [Link]
- **Scout interface**
    - [x] N/A
    - [ ] Updated: [Link]

### Validation criteria

Validation criteria to be added to validation report PR: [LINK-TO-VALIDATION-REPORT-PR from the [validations repository](https://github.com/Clinical-Genomics/validations/)]

**Version specific criteria**

 N/A

> [!IMPORTANT]
> One of the below checkboxes for validation need to be checked 

- [ ] Added version specific validation criteria to validation report
- [ ] Changes validated in standard sections: [validation-section]
- [x] Validation criteria not necessary

## Checklist

> [!IMPORTANT]  
> Ensure that all checkboxes below are ticked before merging.

### For Developers

- **PR Description**
    - [x] Provided a comprehensive description of the PR.
    - [x] Linked relevant user stories or issues to the PR.
- **Documentation**
    - [x] Verified and updated documentation if necessary.
- **Validation criteria**
    - [x] Completed the validation criteria section of the template.
- **Tests**
    - [x] Described and tested the functionality addressed in the PR.
    - [x] Ensured integration of the new code with existing workflows.
    - [x] Confirmed that meaningful unit tests were added for the changes introduced.
    - [x] Checked that the PR has successfully passed all relevant code smells and coverage checks.
- **Review**
    - [ ] Addressed and resolved all the feedback provided during the code review process.
    - [ ] Obtained final approval from designated reviewers.

### For Reviewers

- **Code**
    - [ ] Code implements the intended features or fixes the reported issue.
    - [ ] Code follows the project's coding standards and style guide.
- **Documentation**
    - [ ] Pipeline changes are well-documented in the CHANGELOG and relevant documentation.
- **Validation criteria**
    - [ ] The author has completed the validation criteria section of the template
- **Tests**
    - [ ] The author provided a description of their manual testing, including consideration of edge cases and boundary
      conditions where applicable, with satisfactory results.
- **Review**
    - [ ] Confirmed that the developer has addressed all the comments during the code review.

### CODE REVIEW

**Comment by mathiasbio** [comment]
The same error was mentioned here: https://github.com/brentp/somalier/issues/144

With a fix implemented here: https://github.com/brentp/somalier/commit/867db1966ab8dc18e9c658b4c7164c3a5d7121cf

I will try this out and see if it works

File: CHANGELOG.rst [hunk 1]
```diff
@@ -24,7 +24,8 @@ Removed:
 
 Fixed:
 ^^^^^^
-
+* Merged VarDict and TNscope variants now correctly show both callers in FOUND_IN info field https://github.com/Clinical-Genomics/BALSAMIC/pull/149
```
Comment (fevac) [inline CHANGELOG.rst:27]: This doesn't seem to belong here

**Review by fevac — APPROVED** [review]
💯

File: CHANGELOG.rst (hunk 1 above)
Comment (mathiasbio) [reply CHANGELOG.rst:28]: Yeah, there's been some messy changelog conflicts this round of development 😬
